src/SSLSocket.c

```diff
@@ -552,23 +552,31 @@ int SSLSocket_setSocketForSSL(networkHandles* net, MQTTClient_SSLOptions* opts)
 	if (net->ctx != NULL || (rc = SSLSocket_createContext(net, opts)) == 1)
 	{
 		int i;
-		SSL_CTX_set_info_callback(net->ctx, SSL_CTX_info_callback);
-		SSL_CTX_set_msg_callback(net->ctx, SSL_CTX_msg_callback);
-   		if (opts->enableServerCertAuth) 
-			SSL_CTX_set_verify(net->ctx, SSL_VERIFY_PEER, NULL);
+		if (net->ctx != NULL)
```

Comment:
With the current line 552, SSLSocket_createContext() is only called
if net->ctx is NULL. Otherwise SSLSocket_createContext() creates
a new context and sets net->ctx. But the check that this happened successful
was missing.